cmds/subvolume.c

```diff
@@ -304,13 +289,14 @@ static int cmd_subvolume_create(const struct cmd_struct *cmd, int argc, char **a
 
 	for (int i = optind; i < argc; i++) {
 		ret = create_one_subvolume(argv[i], inherit, create_parents);
-		if (ret < 0)
+		if (ret)
 			has_error = true;
 	}
 	if (!has_error)
 		retval = 0;
 out:
-	free(inherit);
+	if (inherit)
+		btrfs_util_qgroup_inherit_destroy(inherit);
```

Comment:
You can skip the `inherit` check since `free()` can handle NULL pointer without any problem.